core/server/api/settings.js

```diff
@@ -103,6 +103,15 @@ canEditAllSettings = function (settingsInfo, options) {
                 ));
             }
 
+            if (setting.key === 'activeTheme') {
```

Comment:
I do appreciate the desire to have less code - maybe we could also change it so that core settings can only be changed via the model, never the API, but this seems outside the scope of this PR?